postgresql/postgresql_client.go

```diff
@@ -0,0 +1,167 @@
+package postgresql
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/Azure-Samples/azure-sdk-for-go-samples/internal/config"
+	"github.com/Azure-Samples/azure-sdk-for-go-samples/internal/iam"
+	pg "github.com/Azure/azure-sdk-for-go/services/preview/postgresql/mgmt/2020-02-14-preview/postgresqlflexibleservers"
+	"github.com/Azure/go-autorest/autorest"
+	"github.com/Azure/go-autorest/autorest/to"
+)
+
+// GetServersClient returns
+func GetServersClient() pg.ServersClient {
+	serversClient := pg.NewServersClient(config.SubscriptionID())
+	a, _ := iam.GetResourceManagementAuthorizer()
+	serversClient.Authorizer = a
+	serversClient.AddToUserAgent(config.UserAgent())
+	return serversClient
+}
+
+// CreateServer creates a new PostgreSQL Server
+func CreateServer(ctx context.Context, serversClient pg.ServersClient, serverName string, dbLogin string, dbPassword string) (server pg.Server, err error) {
+
+	// Create the server
+	future, err := serversClient.Create(
+		ctx,
+		config.GroupName(),
+		serverName,
+		pg.Server{
+			Location: to.StringPtr(config.Location()),
+			Sku: &pg.Sku{
+				Name: to.StringPtr("Standard_D4s_v3"),
+				Tier: "GeneralPurpose",
+			},
+			ServerProperties: &pg.ServerProperties{
+				AdministratorLogin:         to.StringPtr(dbLogin),
+				AdministratorLoginPassword: to.StringPtr(dbPassword),
+				Version:                    pg.OneTwo,
+				StorageProfile: &pg.StorageProfile{
+					StorageMB: to.Int32Ptr(524288),
+				},
+			},
+		})
+
+	if err != nil {
+		return server, fmt.Errorf("cannot create pg server: %v", err)
+	}
+
+	err = future.WaitForCompletionRef(ctx, serversClient.Client)
+	if err != nil {
+		return server, fmt.Errorf("cannot get the pg server create or update future response: %v", err)
+	}
+
+	return future.Result(serversClient)
+}
+
+// UpdateServerStorageCapacity given the server name and the new storage capacity it updates the server's storage capacity.
+func UpdateServerStorageCapacity(ctx context.Context, serversClient pg.ServersClient, serverName string, storageCapacity int32) (server pg.Server, err error) {

```

Comment:
Actually we already have a function to get the client, could we remove the client from the parameter list, just call the `GetServersClient` function to get the client inside this function?

Reply:
Fixed